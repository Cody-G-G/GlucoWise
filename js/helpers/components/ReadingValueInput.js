'use strict';
import React, {Component} from 'react';
import {TextInput} from 'react-native';

export default class ReadingValueInput extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <TextInput style={this.props.style}
                       value={this.props.inputValue}
                       keyboardType={'numeric'}
                       maxLength={5}
                       onChangeText={(input) => this.props.onChangeText(input)}/>
        );
    }
}