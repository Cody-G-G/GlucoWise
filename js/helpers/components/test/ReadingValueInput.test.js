'use strict';
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import ReadingValueInput from "../ReadingValueInput";

test('ReadingValueInput - renders correctly', () => {
    const style = {
        fontSize: 25,
        color: 'black',
        fontWeight: 'bold',
        flex: 2.5,
        textAlign: 'center'
    };

    const tree = renderer.create(
        <ReadingValueInput style={style}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});