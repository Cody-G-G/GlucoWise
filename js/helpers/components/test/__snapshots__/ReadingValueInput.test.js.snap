// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadingValueInput - renders correctly 1`] = `
<TextInput
  keyboardType="numeric"
  maxLength={5}
  onChangeText={[Function]}
  style={
    Object {
      "color": "black",
      "flex": 2.5,
      "fontSize": 25,
      "fontWeight": "bold",
      "textAlign": "center",
    }
  }
  value={undefined}
/>
`;
